### PR TITLE
Add support for SameSite for cookies

### LINKED ISSUE #31122: RESTEasy Reactive: support NewCookie `sameSite` attribute
### Description

In `main` for Jakarta 3.1 there's support for `NewCookie.SameSite` but our `org.jboss.resteasy.reactive.common.headers.NewCookieHeaderDelegate` does not serialise it, so it doesn't end up in the response.

We should support it.

Also see https://github.com/quarkiverse/quarkus-renarde/issues/100

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
This is a new feature of Jakarta REST Services 3.1

- Closes:  https://github.com/quarkusio/quarkus/pull/31122

### CODE REVIEW

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java [hunk 1]
```diff
@@ -59,6 +60,8 @@ public Object fromString(String newCookie) throws IllegalArgumentException {
                 version = Integer.parseInt(value);
             } else if (name.equalsIgnoreCase("HttpOnly")) {
                 httpOnly = true;
+            } else if (name.equalsIgnoreCase("SameSite")) {
+                sameSite = NewCookie.SameSite.valueOf(value.toUpperCase());
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java:64]: What happens if a value different than the ones in the `SameSite` enum is passed along? The request fails or should we catch here and assume the value is `null`?

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java [hunk 2]
```diff
@@ -71,7 +74,8 @@ public Object fromString(String newCookie) throws IllegalArgumentException {
             cookieValue = "";
         }
 
-        return new NewCookie(cookieName, cookieValue, path, domain, version, comment, maxAge, expiry, secure, httpOnly);
+        return new NewCookie(cookieName, cookieValue, path, domain, version, comment, maxAge, expiry, secure, httpOnly,
+                sameSite);
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java:78]:
```suggestion
        return new NewCookie.Builder(cookieName)
                .value(cookieValue)
                .path(path)
                .domain(domain)
                .version(version)
                .maxAge(maxAge)
                .expiry(expiry)
                .secure(secure)
                .httpOnly(httpOnly)
                .sameSite(sameSite)
                .build()
```

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java:64]: It fails just like Max-age fails. I don't think we should catch the exception (and neither does Jersey)

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java (hunk 2 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/NewCookieHeaderDelegate.java:78]: I propose you push this as a separate commit 😉

**Review by gastaldi — APPROVED** [review]

**Review by FroMage — COMMENTED** [review]
Thanks!

**Comment by geoand** [comment]
🙏🏼

**Pull request merged**